 SystemUI: Fix SystemUI Crash Issue

Add exception handling in systemui to avoid systemui crash

diff --git a/packages/SystemUI/src/com/android/systemui/recents/views/TaskStackViewTouchHandler.java b/packages/SystemUI/src/com/android/systemui/recents/views/TaskStackViewTouchHandler.java
old mode 100644
new mode 100755
index 6000fb5..d93e7dd
--- a/packages/SystemUI/src/com/android/systemui/recents/views/TaskStackViewTouchHandler.java
+++ b/packages/SystemUI/src/com/android/systemui/recents/views/TaskStackViewTouchHandler.java
@@ -636,7 +636,10 @@ private void updateTaskViewTransforms(float dismissFraction) {
             }
 
             int taskIndex = mCurrentTasks.indexOf(task);
-            if (taskIndex == -1) {
+            if (taskIndex == -1 || mCurrentTaskTransforms.size() == 0
+                    || mFinalTaskTransforms.size() == 0
+                    || taskIndex >= mCurrentTaskTransforms.size()
+                    || taskIndex >= mFinalTaskTransforms.size()) {
                 // If a task was added to the stack view after the start of the dismiss gesture,
                 // just ignore it
                 continue;
